shell/platform/darwin/ios/framework/Source/FlutterEnginePlatformViewTest.mm

```diff
@@ -0,0 +1,101 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+
+#import "flutter/fml/message_loop.h"
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterMacros.h"
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngine.h"
+#import "flutter/shell/platform/darwin/ios/platform_view_ios.h"
+
+FLUTTER_ASSERT_NOT_ARC
+
+namespace flutter {
+namespace {
+
+class MockDelegate : public PlatformView::Delegate {
+  void OnPlatformViewCreated(std::unique_ptr<Surface> surface) override {}
+  void OnPlatformViewDestroyed() override {}
+  void OnPlatformViewSetNextFrameCallback(const fml::closure& closure) override {}
+  void OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) override {}
+  void OnPlatformViewDispatchPlatformMessage(fml::RefPtr<PlatformMessage> message) override {}
+  void OnPlatformViewDispatchPointerDataPacket(std::unique_ptr<PointerDataPacket> packet) override {
+  }
+  void OnPlatformViewDispatchSemanticsAction(int32_t id,
+                                             SemanticsAction action,
+                                             std::vector<uint8_t> args) override {}
+  void OnPlatformViewSetSemanticsEnabled(bool enabled) override {}
+  void OnPlatformViewSetAccessibilityFeatures(int32_t flags) override {}
+  void OnPlatformViewRegisterTexture(std::shared_ptr<Texture> texture) override {}
+  void OnPlatformViewUnregisterTexture(int64_t texture_id) override {}
+  void OnPlatformViewMarkTextureFrameAvailable(int64_t texture_id) override {}
+
+  std::unique_ptr<std::vector<std::string>> ComputePlatformViewResolvedLocale(
+      const std::vector<std::string>& supported_locale_data) override {
+    std::unique_ptr<std::vector<std::string>> out = std::make_unique<std::vector<std::string>>();
+    return out;
+  }
+};
+
+}  // namespace
+}  // namespace flutter
+
+@interface FlutterEngine (TestLowMemory)
```

Comment:
Suggestion: Test behavior, not implementation. Per that guidance, the test would check that both the shell and application (on the system platform channel) have the notification propagated to them. OTOH, this test checks an intermediate implementation detail in `FlutterEngine`. This has the following issues:
* If the name of the private method is updated (`notifyLowMemory` is in `FlutterEngine_Internal`), the test breaks even if the implementation is doing the right thing.
* If the implementation changes (say you forgot to send the notification on the system channel in an update), the test still passes but the behavior has changed meaningfully.

Reply:
I agree this would be better - I had been avoiding it because it's harder to test.

I have an idea about how to make this more testable - we could give the engine a delegate for shell creation so that tests can inject their own. I'll open a separate patch with an approach for that to see if it makes sense and not have it get caught up in this.